python/traverse.py

```diff
@@ -0,0 +1,63 @@
+from typing import Callable, Union
+
+import lxml.html
+
+
+def traverse_and_execute(
+    html: str, fn: Callable[[lxml.html.HtmlElement], None]
+) -> None:
+    elements = lxml.html.fragments_fromstring(html)
+    root_parent = elements[0].getparent()
+    for element in root_parent:
+        traverse_and_execute_impl(element, fn)
+    return root_parent.getchildren()
+
+
+def traverse_and_execute_impl(
+    element: lxml.html.HtmlElement, fn: Callable[[lxml.html.HtmlElement], None]
+) -> None:
+    new_element = fn(element)
+    has_new_element = (
+        new_element is not None
+        and new_element is not element
```

Comment:
Why use the "is not" comparison here instead of "!="? I don't think we want to check whether these are equivalent objects in memory.

Reply:
That is actually exactly what I want to check; this supports returning the element that was passed to this function because I thought that might be something I need. I don't know if the current code relies on that though; I can reevaluate.